lib/aws/http_client.ex

```diff
@@ -23,7 +23,14 @@ defmodule AWS.HTTPClient do
     body :: iodata(),
     headers :: list(),
     options :: keyword()
-  ) :: {:ok, %{status_code: integer(), body: map()}} | {:error, term()}
+  ) :: {:ok, %{status_code: integer(), headers: list({binary(), binary()}), body: map()}} | {:error, term()}
```

Comment:
nitpick: we can replace `list(t)` with a shorter `[t]` but more importantly, I think `body` is a `binary()` a not a `map()`.

```suggestion
  ) :: {:ok, %{status_code: integer(), headers: [{binary(), binary()}], body: binary()}} | {:error, term()}
```